tom_lt/lt.py

```diff
@@ -272,26 +456,64 @@ def get_form(self, observation_type):
             return LT_IOO_ObservationForm
 
     def submit_observation(self, observation_payload):
-        print("HERE")
-        headers = {
-            'Username': LT_SETTINGS['username'],
-            'Password': LT_SETTINGS['password']
-        }
-        url = '{0}://{1}:{2}/node_agent2/node_agent?wsdl'.format('http', LT_HOST, LT_PORT)
-        client = Client(url=url, headers=headers)
-
-        response = client.service.handle_rtml(observation_payload)
-        response_rtml = etree.fromstring(response)
-        obs_id = response_rtml.get('uid').split('-')[-1]
-        return [obs_id]
+        if(DEBUG):
+            payload = etree.fromstring(observation_payload)
+            f = open("django.RTML", "w")
+            f.write(etree.tostring(payload, encoding="unicode", pretty_print=True))
+            f.close()
+            return [0]
+        else:
+            headers = {
+                'Username': LT_SETTINGS['username'],
+                'Password': LT_SETTINGS['password']
+            }
+            url = '{0}://{1}:{2}/node_agent2/node_agent?wsdl'.format('http', LT_HOST, LT_PORT)
+            client = Client(url=url, headers=headers)
+            # Send payload, and receive response string, removing the encoding tag which causes issue with lxml parsing
+            response = client.service.handle_rtml(observation_payload).replace('encoding="ISO-8859-1"', '')
+            response_rtml = etree.fromstring(response)
+            mode = response_rtml.get('mode')
+            if mode == 'reject':
+                print(response)
+            obs_id = response_rtml.get('uid')
+            return [obs_id]
 
     def cancel_observation(self, observation_id):
         form = self.get_form()()
         payload = form._build_prolog()
         payload.append(form._build_project())
 
-    def validate_observation(self, observation_payload):
-        return
+    def validate_observation(observation_payload):
```

Comment:
I realize that it doesn't make a functional difference to use `self`, but for consistency's sake with the other modules, I would encourage it. Also, because it's a method on the interface you're implementing, that means if we start calling it from the core TOM Toolkit, the LT module won't break.

Reply:
Kind of fixed, though I don't fully understand why calling the following from ```is_valid()```,

```python
errors = LTFacility.validate_observation(self.observation_payload())
```

gives the following error, which is confusing when, for example, the Gemini module is only passing the Observation payload when it is doing the ```is_valid()``` checks ;
```
validate_observation() missing 1 required positional argument: 'observation_payload'
```

Have updated the the call to be ``` LTFacility.validate_observation(self, self.observation_payload())``` and seems to work fine.